lib/model/content.dart

```diff
@@ -872,6 +870,53 @@ class _ZulipContentParser {
     return descendant4.text.trim();
   }
 
+  UserMentionNode? parseUserMention(dom.Element element) {
+    assert(_debugParserContext == _ParserContext.inline);
+    assert(element.localName == 'span');
+    final debugHtmlNode = kDebugMode ? element : null;
+
+    final classes = element.className.split(' ')..sort();
+    assert(classes.contains('user-mention')
+        || classes.contains('user-group-mention'));
+    int i = 0;
+
+    if (i >= classes.length) return null;
+    bool isChannelWildcardClassIncluded = false;
+    if (classes[i] == 'channel-wildcard-mention') {
+      // A newly-added class for channel wildcard mentions.
+      // See: https://github.com/zulip/zulip/pull/31075
+      i++;
+      isChannelWildcardClassIncluded = true;
+    }
+
+    if (i >= classes.length) return null;
+    if (classes[i] == 'silent') {
+      // A silent @-mention.  We ignore this flag; see [UserMentionNode].
+      i++;
+    }
+
+    if (i >= classes.length) return null;
+    if (classes[i] == 'user-mention' ||
+       (classes[i] == 'user-group-mention' && !isChannelWildcardClassIncluded)) {
```

Comment:
nit: indentation; and operators go after newline:
```suggestion
    if (classes[i] == 'user-mention'
        || (classes[i] == 'user-group-mention' && !isChannelWildcardClassIncluded)) {
```

(See examples in `parseInlineContent` below.)

With less indentation, the second sub-condition looks like it's a peer of the `if` condition as a whole.

For the rationale for `||` after newline, see:
https://github.com/zulip/zulip-mobile/blob/a115df1f71c9dc31e9b41060a8d57b51c017d786/tools/formatting.eslintrc.yaml#L20-L23